libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1404,45 +1425,85 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 
 	ABIFunctions abi(m_context.evmVersion(), m_context.revertStrings(), m_context.functionCollector());
 
-	solUnimplementedAssert(!funType.isBareCall(), "");
 	Whiskers templ(R"(
 		<?checkExistence>
 			if iszero(extcodesize(<address>)) { revert(0, 0) }
 		</checkExistence>
 
 		// storage for arguments and returned data
 		let <pos> := <freeMemory>
-		mstore(<pos>, <shl28>(<funId>))
-		let <end> := <encodeArgs>(add(<pos>, 4) <argumentString>)
+		<?bareCall>
+		<!bareCall>
+			mstore(<pos>, <shl28>(<funId>))
+		</bareCall>
+		let <end> := <encodeArgs>(
+			<?bareCall>
+				<pos>
+			<!bareCall>
+				add(<pos>, 4)
+			</bareCall>
+			<argumentString>
+		)
 
 		let <success> := <call>(<gas>, <address>, <?hasValue> <value>, </hasValue> <pos>, sub(<end>, <pos>), <pos>, <reservedReturnSize>)
-		<?noTryCall>
+		<?revertOnFailure>
 			if iszero(<success>) { <forwardingRevert>() }
-		</noTryCall>
+		</revertOnFailure>
 		<?hasRetVars> let <retVars> </hasRetVars>
-		if <success> {
-			<?dynamicReturnSize>
-				// copy dynamic return data out
-				returndatacopy(<pos>, 0, returndatasize())
-			</dynamicReturnSize>
-
-			// update freeMemoryPointer according to dynamic return size
-			mstore(<freeMemoryPointer>, add(<pos>, <roundUp>(<returnSize>)))
-
-			// decode return parameters from external try-call into retVars
-			<?hasRetVars> <retVars> := </hasRetVars> <abiDecode>(<pos>, add(<pos>, <returnSize>))
-		}
+		<?returnSuccessAndReturndata>
+			<?returnReturndata> let <returndataVar> := <extractReturndataFunction>() </returnReturndata>
+		<!returnSuccessAndReturndata>
+			if <success> {
+				<?dynamicReturnSize>
+					// copy dynamic return data out
+					returndatacopy(<pos>, 0, returndatasize())
+				</dynamicReturnSize>
+
+				// update freeMemoryPointer according to dynamic return size
+				mstore(<freeMemoryPointer>, add(<pos>, <roundUp>(<returnSize>)))
+
+				// decode return parameters from external try-call into retVars
+				<?hasRetVars> <retVars> := </hasRetVars> <abiDecode>(<pos>, add(<pos>, <returnSize>))
+			}
+		</returnSuccessAndReturndata>
```

Comment:
This is probably what CI is complaining about:
```suggestion
		<?returnSuccessConditionAndReturndata>
			<?returnReturndata> let <returndataVar> := <extractReturndataFunction>() </returnReturndata>
		<!returnSuccessConditionAndReturndata>
			if <success> {
				<?dynamicReturnSize>
					// copy dynamic return data out
					returndatacopy(<pos>, 0, returndatasize())
				</dynamicReturnSize>

				// update freeMemoryPointer according to dynamic return size
				mstore(<freeMemoryPointer>, add(<pos>, <roundUp>(<returnSize>)))

				// decode return parameters from external try-call into retVars
				<?hasRetVars> <retVars> := </hasRetVars> <abiDecode>(<pos>, add(<pos>, <returnSize>))
			}
		</returnSuccessConditionAndReturndata>
```